tests/data/dataset_readers/penn_tree_bank_reader_test.py

```diff
@@ -47,7 +47,8 @@ def test_read_from_file(self):
                                     "VP(TO to)(VP(VB be)(ADJP(JJ fair)(PP(TO to)(NP(JJ other)(NNS "
                                     "bidders))))))))))))))(. .)))")
 
-        assert fields["gold_tree"].metadata == gold_tree
+        assert fields["metadata"].metadata["gold_tree"] == gold_tree
+        assert fields["metadata"].metadata["tokens"] == tokens
```

Comment:
this always feels clunky to me, I wonder if `MetadataField` should implement `__getitem__`? (not in this PR of course, just in general)

Reply:
We could do this for all of our fields, e.g. `for token in TextField` etc, etc. I'm a fan

Reply:
What does `for token in TextField` return, and when is it called?

For the `MetadataField`, it was originally a hack to get something working for BiDAF.  There is definitely room to improve it.

Reply:
https://github.com/allenai/allennlp/blob/master/tests/data/dataset_readers/penn_tree_bank_reader_test.py#L24

it would be equivalent to that list comprehension (or probably returning `Tokens`). Maybe it's less useful because it's unclear what it would return, but it would make it much cleaner.